aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,170 @@
+
+Components can be:
+
+* Nested.
+* Generated procedurally in code.
+* Shared via class libraries.
+* Unit tested without requiring a browser DOM.
+
+## Infrastructure
+
+Blazor offers the core facilities that most apps require, including:
+
+* Layouts
+* Routing
+* Dependency injection
+* Lazy loading (loading parts of the app on demand as a user navigates the app)
+* Unit testing
+
+As an important design point, all of these features are optional. When one of these features isn't used in an app, the implementation is stripped out of the app when published by the IL linker.
+
+Only a few low-level elements are baked into the framework. For example, routing and layouts aren't built-in. Routing and layouts are implemented in *user space*, code that an app developer can write without using internal APIs. These features can be replaced with different systems to suit the app's requirements. The current layouts prototype is implemented in only about 30 lines of C# code, so a developer can easily understand and replace it if desired.
+
+## Code sharing and .NET Standard
+
+The [.NET Standard](/dotnet/standard/net-standard) is a formal specification of .NET APIs that are intended to be available on all .NET implementations. Mono on WebAssembly supports `netstandard2.0` or higher. .NET Standard class libraries can be shared across backend server code and in browser-based apps.
+
+Browsers support the APIs that developers use to build web apps. Not all .NET APIs make sense in the browser. For example, arbitrary TCP sockets can't be accessed in a browser, so [System.Net.Sockets.TcpListener](/dotnet/api/system.net.sockets.tcplistener) can't perform any useful task. For BCL APIs that don't apply to a given platform, the BCL throws a [PlatformNotSupported](/dotnet/api/system.platformnotsupportedexception) exception.
+
+## JavaScript/TypeScript interop
+
+Sometimes, access to third-party JavaScript libraries and browser APIs is required. WebAssembly is designed to interoperate with JavaScript, so Blazor is capable of using any library or API that JavaScript is able to use.
+
+To call JavaScript libraries or your own custom JavaScript/TypeScript code from .NET, the current approach is to register a named function in a JavaScript/TypeScript file:
+
+```javascript
+Blazor.registerFunction('doPrompt', message => {
+  return prompt(message);
+});
+```
+
+Wrap the named function for calls from .NET:
+
+```csharp
+public static bool DoPrompt(string message)
+{
+    return RegisteredFunction.Invoke<bool>("doPrompt", message);
+}
+```
+
+This approach has the benefit of working with JavaScript build tools, such as [webpack](https://webpack.js.org/).
+
+To save the trouble of registering functions in this way for standard browser APIs, the Mono team is working on a library that exposes standard browser APIs to .NET.
```

Comment:
Consider:
The Mono team is working on a library that exposes standard browser APIs to .NET.
That's probably enough - but you could tack on, so writing wrappers ...